superset/assets/src/dashboard/components/Dashboard.jsx

```diff
@@ -247,14 +247,17 @@ class Dashboard extends React.PureComponent {
     const slice_id = getModalSliceIDFor(this.props.dashboardState.publishSubscriberMap, filterKey);
     if (slice_id) {
       // open modal with chart in loading state
-      this.updateModalProps(slice_id, true,'loading')
+      this.updateModalProps(slice_id, true, 'loading')
     }
   }
 
-  updateModalProps(slice_id, showModal,status = undefined) {
+  updateModalProps(slice_id, showModal, status = undefined) {
     this.showModal = showModal
     this.modalChart = this.props.charts[slice_id];
     if (this.modalChart) {
+      if (status) {
+        this.modalChart.chartStatus = status
+      }
       this.modalChartStatus = status ? status : this.modalChart.chartStatus;
```

Comment:
We can remove variable modalChartStatus and use directly this.modalChart.chartStatus or directly assign this.modalChart.chartStatus value in this.modalChartStatus variable.

Reply:
done